src/deluge/model/song/song.cpp

```diff
@@ -5732,6 +5750,64 @@ int32_t Song::addInstrumentsToFileItems(OutputType outputType) {
 	return NO_ERROR;
 }
 
+void Song::displayCurrentRootNoteAndScaleName() {
+	DEF_STACK_STRING_BUF(popupMsg, 40);
+	char noteName[5];
+	int32_t isNatural = 1; // gets modified inside noteCodeToString to be 0 if sharp.
+	noteCodeToString(currentSong->rootNote, noteName, &isNatural);
+
+	popupMsg.append(noteName);
+	if (display->haveOLED()) {
+		popupMsg.append(" ");
+		popupMsg.append(getScaleName(getCurrentPresetScale()));
+	}
+	display->displayPopup(popupMsg.c_str());
+}
+
+void Song::transpose(int32_t offset) {
+	if (anyScaleModeClips()) {
+		if (masterTransposeOffset != 0) {
+			offset *= currentSong->masterTransposeOffset;
+		}
+		transposeAllScaleModeClips(offset);
+	}
+	else {
+		display->displayPopup(deluge::l10n::get(deluge::l10n::String::STRING_FOR_CANT_TRANSPOSE));
+	}
+}
+
+void Song::adjustMasterTransposeOffset(int32_t offset) {
+	masterTransposeOffset += offset;
+	if (masterTransposeOffset < 0) {
+		masterTransposeOffset = 0;
+	}
+	displayMasterTransposeOffset();
+}
+
+void Song::displayMasterTransposeOffset() {
+	DEF_STACK_STRING_BUF(popupMsg, 40);
+
+	if (display->haveOLED()) {
+		popupMsg.append("Transpose Offset: \n");
```

Comment:
should be - "Transpose Interval: " rather than offset. Offset sounds like they should be added, interval is clearer that you're changing the interval that the transposer will adjust by

Same applies to the names in the code